src/ocrd_utils/os.py

```diff
@@ -186,13 +260,16 @@ def get_processor_resource_types(executable, ocrd_tool=None):
         if not which(executable):
             return ['*/*']
         ocrd_tool = get_ocrd_tool_json(executable)
-    if not next((True for p in ocrd_tool.get('parameters', {}).values() if 'content-type' in p), False):
-        # None of the parameters for this processor are resources (or not
-        # the resource parameters are not properly declared, so output both
-        # directories and files
+    mime_types = [mime
+                  for param in ocrd_tool['parameters'].values()
+                  if param['type'] == 'string' and param.get('format', '') == 'uri' and 'content-type' in param
+                  for mime in param['content-type'].split(',')]
```

Comment:
Why `ocrd_tool['parameters']` instead of the safer `ocrd_tool.get('parameters', {})`?

Reply:
I was under the false impression that we can rely on OcrdToolValidator generating these as defaults. But what we actually get is the raw dict from the JSON, so you are right, it should be that